jormungandr/src/settings/logging.rs

```diff
@@ -151,37 +152,51 @@ impl LogSettings {
 impl LogSettingsEntry {
     fn to_subscriber(
         &self,
-    ) -> Result<
-        (
-            Box<dyn Subscriber + Send + Sync>,
-            Option<tracing_appender::non_blocking::WorkerGuard>,
-        ),
-        Error,
-    > {
+    ) -> Result<(Box<dyn Subscriber + Send + Sync>, Option<WorkerGuard>), Error> {
         let Self {
             output,
             level,
             format,
         } = &self;
 
-        let builder = tracing_subscriber::fmt::SubscriberBuilder::default();
+        let builder = SubscriberBuilder::default();
+
+        fn build_writer_subscriber(
+            builder: SubscriberBuilder,
+            writer: impl Write + Send + Sync + 'static,
+            level: LevelFilter,
+            format: LogFormat,
+        ) -> (Box<dyn Subscriber + Send + Sync>, Option<WorkerGuard>) {
+            let (subscriber, guard) = tracing_appender::non_blocking(writer);
+            let builder = builder.with_writer(subscriber).with_max_level(level);
+            let subscriber: Box<dyn Subscriber + Send + Sync> = match format {
+                LogFormat::Plain => Box::new(builder.finish()),
+                LogFormat::Json => Box::new(builder.json().finish()),
+            };
+            (subscriber, Some(guard))
+        }
```

Comment:
Lovely. Everything was separated at some point in the refactor. But the problems I had making the dynamic dispatching work made me use excessive duplicated code indeed. Thanks!